### PR TITLE
adds a new data source tailscale_4via6

### PR DESCRIPTION
**Pull request by MrGossett**
**What this PR does / why we need it**:

Adds a new data source `tailscale_4via6` to generate IPv6 routes for a given site ID and IPv4 route. The data source implements the same logic as running `tailscale debug via <site> <cidr>`.

This will allow Tailscale users testing the [4via6](https://tailscale.com/kb/1201/4via6-subnets/) feature to generate IPv6 routes in Terraform.

Example:

```hcl
data "tailscale_4via6" "network" {
  site = 7
  cidr = "10.1.1.0/24"
  # ipv6 will be "fd7a:115c:a1e0:b1a:0:7:a01:100/120"
}

data "tailscale_device" "router" {
  name = "4via6-router.example.com"
}

resource "tailscale_device_subnet_routes" "4via6" {
  device_id = data.tailscale_device.router.id
  routes    = [data.tailscale_4via6.network.ipv6]
}
```

**Special notes for your reviewer**:

I recognize that the 4via6 feature is still in alpha and is only available to Business plan users. I believe having the data source available will be valuable to the set of customers who are managing their Tailnet with Terraform, are on the Business plan, and want to test the 4via6 feature.

### CODE REVIEW

File: docs/data-sources/4via6.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
```
Comment (davidsbond) [inline docs/data-sources/4via6.md:1]: This documentation is lacking the example usage so it seems you may need to regenerate the documentation or double check you've created the example file correctly.

File: tailscale/data_source_4via6.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"inet.af/netaddr"
+	"tailscale.com/net/tsaddr"
+)
+
+func dataSource4Via6() *schema.Resource {
+	return &schema.Resource{
+		Description: "Calculate a 4via6 address",
```
[the remark this answers is not in the document]
Comment (MrGossett) [reply tailscale/data_source_4via6.go]: 👍🏻 `a173b1b`

File: docs/data-sources/4via6.md (hunk 1 above)
Comment (MrGossett) [reply docs/data-sources/4via6.md:1]: 👍🏻 `a173b1b`

File: tailscale/data_source_4via6.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+			},
+		},
+	}
+}
+
+func dataSource4Via6Read(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	site := uint32(d.Get("site").(int))
+	cidr, err := netaddr.ParseIPPrefix(d.Get("cidr").(string))
+	if err != nil {
+		return diagnosticsError(err, "invalid CIDR")
```
[the remark this answers is not in the document]
Comment (MrGossett) [reply tailscale/data_source_4via6.go]: 👍🏻 `a173b1b`

**Comment by MrGossett** [comment]
Are there any other changes that should be addressed before this is merged?

**Comment by davidsbond** [comment]
Hey @MrGossett, this looks fine to me although we're still in the process of migrating everything for this terraform provider from my personal GitHub & Terraform registry namespace to Tailscale's. So it might be a little while until someone from Tailscale can put eyes on this PR.

**Comment by MrGossett** [comment]
@davidsbond understood, thanks for clarifying. I just wanted to make sure nothing was pending action from me :)

**Review by DentonGentry — APPROVED** [review]

**Comment by MrGossett** [comment]
Thank you, @DentonGentry! 🎉

**Pull request merged**